### ISSUE TITLE
Rule suggestion: List style (brackets)

### ISSUE THREAD

**Issue by carlpett** [adrienverge/yamllint#236]
I'd like to propose a rule for checking style for lists. We have some files where we mix the two forms:
```yaml
foo: ["a","b","c"]
bar:
  - 1
  - 2
  - 3
```

Having options to only allow either, or consistent would be great!

_(To some extent also dicts, where we sometimes get `baz: { qux: true }`, but this seems less common)_

**Comment by adrienverge** [comment]
Hey, this is a good idea.

- This could be done by adding some options to [braces](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.braces) and [brackets](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.brackets), for example:
  ```yaml
  brackets:
    forbid: true
  ```
  Drawback: we cannot force the *flow* style (the one with brackets) and forbid the *block* style (the multi-line one).
  
- Another option would be to add 2 other rules:
  ```yaml
  mappings-style:
    type: block | flow | whatever
  sequences-style:
    type: block | flow | whatever
  ```

**Comment by make-github-pseudonymous-again** [comment]
How difficult would it be to implement this?
